docker-compose.yml

```diff
@@ -0,0 +1,63 @@
+version: '2'
+
+services:
+  db:
+    image: 'postgres:latest'
+    container_name: db
+    ports:
+      - "5432:5432"
+    environment:
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=postgres
+      - POSTGRES_DB=acmestore
+      
+  persist:
+    image: 'acme-store-persist:latest'
+    build:
+      context: /workspace/acme-store-management/acme-store-persist
+      dockerfile: /workspace/acme-store-management/acme-store-persist/Dockerfile
+    container_name: persist
+    ports:
+      - "8082:8082"
+    depends_on:
+      - db
+    environment:
+      - SPRING_DATASOURCE_URL=jdbc:postgresql://db:5432/acmestore
+      - SPRING_DATASOURCE_USERNAME=postgres
+      - SPRING_DATASOURCE_PASSWORD=postgres
+      - SPRING_JPA_HIBERNATE_DDL_AUTO=create
+          
+  import:
+    image: 'acme-store-import:latest'
+    build:
+      context: /workspace/acme-store-management/acme-store-import
+      dockerfile: /workspace/acme-store-management/acme-store-import/Dockerfile
+    container_name: import
+    ports:
+      - "8087:8087"
+    depends_on:
+      - persist
+      
+  react-frontend:
+    image: 'react-acme-store:latest'
+    build:
+      context: /workspace/acme-store-management/react-acme-store
+      dockerfile: /workspace/acme-store-management/react-acme-store/Dockerfile
+    container_name: react-frontend
+    command: npm start
+    ports:
+      - "3000:3000"
+    depends_on:
+      - persist
+      
+      
+  vue-frontend:
+    image: 'vue-acme-store:latest'
+    build:
+      context: /workspace/acme-store-management/vue-acme-store
+      dockerfile: /workspace/acme-store-management/vue-acme-store/Dockerfile
+    container_name: vue-frontend
+    ports:
+      - "8081:8081"
+    depends_on:
+      - persist
```

Comment:
```
version: '2'

services:
  db:
    image: 'postgres:latest'
    container_name: db
    ports:
      - "5432:5432"
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
      - POSTGRES_DB=acmestore
      
  backend:
    build: .
    container_name: persist
    ports:
      - "8082:8082"
    depends_on:
      - db
    environment:
      - SPRING_DATASOURCE_URL=jdbc:postgresql://db:5432/acmestore
      - SPRING_DATASOURCE_USERNAME=postgres
      - SPRING_DATASOURCE_PASSWORD=postgres
      - SPRING_JPA_HIBERNATE_DDL_AUTO=create
          
  react-frontend:
    image: 'react-acme-store:latest'
    build: ./react-acme-store
    container_name: react-frontend
    command: npm start
    ports:
      - "3000:3000"
    depends_on:
      - backend
      
      
  vue-frontend:
    image: 'vue-acme-store:latest'
    build: ./vue-acme-store
    container_name: vue-frontend
    ports:
      - "8081:8081"
    depends_on:
      - backend
```

Reply:
This should work if you also add the following Dockerfile to the project root:

```
FROM adoptopenjdk:15-jre-openj9

RUN apt-get update
RUN apt-get install -y maven tree

WORKDIR /app


ADD ./pom.xml ./pom.xml
ADD ./acme-store-import/pom.xml ./acme-store-import/pom.xml
ADD ./acme-store-persist/pom.xml ./acme-store-persist/pom.xml

RUN mvn install -Dmaven.repo.local=/root/.m2


ADD ./acme-store-import ./acme-store-import
ADD ./acme-store-persist ./acme-store-persist

RUN mvn install -DskipTests -o -Dmaven.repo.local=/root/.m2


ENTRYPOINT ["mvn", "spring-boot:run", "-Dmaven.repo.local=/root/.m2"]
```

Reply:
Although, it seems the project is missing some modules in this PR? The API and the Commons modules. Can you confirm plz?

```
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] Child module /app/acme-store-api of /app/pom.xml does not exist @ 
[ERROR] Child module /app/acme-store-library-commons of /app/pom.xml does not exist @ 
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project com.acme.store.management:acme-store-management:1.0.0 (/app/pom.xml) has 2 errors
[ERROR]     Child module /app/acme-store-api of /app/pom.xml does not exist
[ERROR]     Child module /app/acme-store-library-commons of /app/pom.xml does not exist
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
The command '/bin/sh -c mvn install -Dmaven.repo.local=/root/.m2' returned a non-zero code: 1
ERROR: Service 'backend' failed to build 
```